### ISSUE TITLE
Facebook / Api Adapter

### ISSUE THREAD

**Issue by Webranger2** [ioBroker/AdapterRequests#414]
Hallo zusammen,

warum gibt es eigentlich noch keinen Adapter für die Facebook API
https://developers.facebook.com/docs/graph-api/

Ich und einige Leute mehr haben sicherlich Facebook Seiten oder Gruppen wo man gerne einige Daten für im iobroker haben möchte.

Mir würden sogar die Standart Zahlen reichen wie likes, neue und verlorene likes usw.

Aber das ist sicherlich dann noch ausbaubar.
Leider bin ich nicht in der Lage so einen Adapter zu schreiben.

Gibt aber bestimmt mehr Leute die sowas interessieren könnte.

Würde mich freuen wenn ein paar Leute zusammen kommen die das interessiert sodas sich die Entwicklung eines solchen adapters lohnt.

Gruss Danyel

Suchergebnisse
Webergebnisse

Dieses Bild dient nur der Aufmerksamkeit...
![Bildschirmfoto 2020-05-24 um 17 39 01](https://user-images.githubusercontent.com/44174579/82758228-856ff500-9de5-11ea-9925-7aa39273f8a8.png)

**Comment by klein0r** [comment]
Einfacher wäre es wahrscheinlich, einen API Key zu generieren und dann einfach mit einem kurzen Blockly-Request die Daten von der Graph API abzuholen. Man braucht ja nicht direkt einen Adapter dafür 😄 Gerade, wenn es nur um einzelne Daten geht.

**Comment by Webranger2** [comment]
Hast du nen Beispiel wie ich das mit Blockly umsetzte? 

Ein Adapter würde es evtl für alle die nicht wissen wie es mit Blockly geht oder einblick in die API haben einfacher machen.

**Comment by klein0r** [comment]
Siehe https://developers.facebook.com/docs/graph-api/reference/v7.0/object/likes

1. https://developers.facebook.com/apps/ Neue App erstellen
2. Unter "Settings > Basics" App ID und App Secret raussuchen
3. Access Token generieren (beides natürlich ersetzen in der URL)

```
curl -X GET "https://graph.facebook.com/oauth/access_token?client_id={your-app-id}&client_secret={your-app-secret}&grant_type=client_credentials"
```

4. Mit dem generierten Token die Daten abfragen. Dafür brauchst Du die PageID (findest Du auf der "About"/"Über" Seite ganz unten. z.B. 1212727978744889

```
https://graph.facebook.com/v7.0/1212727978744889/likes?access_token=xxxx
```

**ABER**: Das geht erstmal nicht mit neuen Apps. Um Live-Daten abzufragen, brauchst Du eine App, welche den Review-Process durchlaufen hat. Siehe unter "App Review -> Permission and Features" die Berechtigung **Page Public Metadata Access**.

Und um eine App in den Review Prozess geben zu können, muss man Dokumente hochladen usw. Also diese Schritte wird wohl kaum jemand gehen.

**Comment by Webranger2** [comment]
Genau aus dem Grund finde ich einen Adapter richtig sinnvoll!

Daten eintragen wie App ID, App Secret und PageID rest macht der Adapter und schreibt ausgewählte Daten in die Objekte.

**Comment by klein0r** [comment]
Ja, aber dann müsste trotzdem jeder User eine App bei Facebook registrieren und vor allem durch den Review Prozess bekommen. Und das ist eine ziemlich große Hürde...

**Comment by klein0r** [comment]
```
Currently Ineligible for Submission
Your submission is missing data in the following fields:
App Icon (1024 x 1024)
Privacy Policy URL
Category
Business Use
```
